### PR TITLE
set othttp span status from HTTP status code

### PR DESCRIPTION
**Pull request by almariah**
This PR fixes a small issue with othttp handler. The handler doesn't set the span status code after serving the HTTP request.

### CODE REVIEW

File: instrumentation/othttp/handler.go [hunk 1]
```diff
@@ -120,6 +120,7 @@ func (h *Handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	h.handler.ServeHTTP(rww, r.WithContext(ctx))
 
 	setAfterServeAttributes(span, bw.read, rww.written, rww.statusCode, bw.err, rww.err)
+	span.SetStatus(standard.SpanStatusFromHTTPStatusCode(int(rww.statusCode)))
```
Comment (Aneurysm9) [inline instrumentation/othttp/handler.go:123]:
```suggestion
	span.SetStatus(standard.SpanStatusFromHTTPStatusCode(rww.statusCode))
```

The cast to `int` isn't necessary as the `rww.statusCode` field is already an `int`.

Comment (MrAlias) [inline instrumentation/othttp/handler.go:123]: Should this be moved into the "post serve" function `setAfterServeAttributes`?

`rww.statusCode` is already an `int` no need for the conversion.

Comment (Aneurysm9) [reply instrumentation/othttp/handler.go:123]: > Should this be moved into the "post serve" function setAfterServeAttributes?

I thought about that as well, but it explicitly says it's setting attributes. v0v  I suppose the name could change or we could ignore the slight mismatch, doesn't bother me much.

Comment (almariah) [reply instrumentation/othttp/handler.go:123]: I thought about that too. But as it's setting attributes I changed it be outside `setAfterServeAttributes`, I guess it makes sense. If needed to be within `setAfterServeAttributes` I can change it.

**Pull request merged**